pkg/arvo/sur/chat-store.hoon

```diff
@@ -0,0 +1,47 @@
+|%
+::
+++  serial  @uvH
+::
++$  letter
+  $%  [%text text=cord]
+      [%url url=cord]
+      [%code expression=cord output=(list tank)]
+  ==
+::
++$  envelope
+  $:  uid=serial
+      number=@
+      author=ship
+      when=time
+      =letter
+  ==
+::
++$  config
+  $:  owner=ship
+      length=@
+      read=@
+  ==
+::
++$  mailbox
+  $:  =config
+      envelopes=(list envelope)
```

Comment:
I'm a little surprised this is a list rather than a map. What's the purpose of the serial uid, and how does it relate to the position on an envelope in the mailbox list?

Reply:
The serial UID makes it much easier for us to build client features such as de-duplication of pending messages, and a list seems like a natural fit for this ordered data structure.